src/main/rx/nrf24_cx10.c

```diff
@@ -0,0 +1,369 @@
+            if (bindPacket) {
+                // set the hopping channels as determined by the txId received in the bind packet
+                cx10SetHoppingChannels(txId);
+                ret = NRF24_RECEIVED_BIND;
+                protocolState = STATE_ACK;
+                ackCount = 0;
+            }
+        }
+        break;
+    case STATE_ACK:
+        // transmit an ACK packet
+        ++ackCount;
+        // send out an ACK on the bind channel, required by deviationTx
+        payload[9] = 0x01;
+        NRF24L01_SetChannel(CX10_RF_BIND_CHANNEL);
+        XN297_WritePayload(payload, payloadSize);
+        NRF24L01_SetTxMode();// enter transmit mode to send the packet
+        NRF24L01_SetRxMode();// and then reenter receive mode
+        // send out an ACK on each of the hopping channels, required by CX10 transmitter
+        for (uint8_t ii = 0; ii < RF_CHANNEL_COUNT; ++ii) {
+            NRF24L01_SetChannel(rfChannels[ii]);
+            XN297_WritePayload(payload, payloadSize);
+            NRF24L01_SetTxMode();// enter transmit mode to send the packet
+            NRF24L01_SetRxMode();// and then reenter receive mode
+        }
+        if (ackCount > ACK_TO_SEND_COUNT) {
+            NRF24L01_SetChannel(rfChannels[0]);
+            // and go into data state to wait for first data packet
+            protocolState = STATE_DATA;
+        }
+        break;
+    case STATE_DATA:
+        if (micros() > timeOfLastHop + hopTimeout) {
+            hopToNextChannel();
+            timeOfLastHop = micros();
+        }
+        // read the payload, processing of payload is deferred
+        if (NRF24L01_ReadPayloadIfAvailable(payload, payloadSize + 2)) {
+            XN297_UnscramblePayload(payload, payloadSize + 2);
+            hopToNextChannel();
+            ret = NRF24_RECEIVED_DATA;
+        }
+    }
+    return ret;
+}
+
+void cx10Init(const rxConfig_t *rxConfig, rxRuntimeConfig_t *rxRuntimeConfig)
+{
+    rxRuntimeConfig->channelCount = CX10_RC_CHANNEL_COUNT;
+    cx10Nrf24Init(rxConfig->nrf24rx_protocol);
+}
+
+
+// XN297 emulation layer
+
+static int xn297_addrLen;
+static uint8_t  xn297_txAddr[5];
+static uint8_t  xn297_rxAddr[5];
+static uint8_t  xn297_crc = 1;
+static const uint8_t xn297_scramble[] = {
+    0xe3, 0xb1, 0x4b, 0xea, 0x85, 0xbc, 0xe5, 0x66,
+    0x0d, 0xae, 0x8c, 0x88, 0x12, 0x69, 0xee, 0x1f,
+    0xc7, 0x62, 0x97, 0xd5, 0x0b, 0x79, 0xca, 0xcc,
+    0x1b, 0x5d, 0x19, 0x10, 0x24, 0xd3, 0xdc, 0x3f,
+    0x8e, 0xc5, 0x2f};
+
+static const uint16_t xn297_crcXorout[] = {
+    0x0000, 0x3448, 0x9BA7, 0x8BBB, 0x85E1, 0x3E8C, // 1st entry is missing, probably never needed
+    0x451E, 0x18E6, 0x6B24, 0xE7AB, 0x3828, 0x8148, // it's used for 3-byte address w/ 0 byte payload only
+    0xD461, 0xF494, 0x2503, 0x691D, 0xFE8B, 0x9BA7,
+    0x8B17, 0x2920, 0x8B5F, 0x61B1, 0xD391, 0x7401,
+    0x2138, 0x129F, 0xB3A0, 0x2988};
+
+static uint8_t bitReverse(uint8_t bIn)
+{
+    uint8_t bOut = 0;
```

Comment:
Interesting. Here are build sizes with 3 variants:

ReverseBits7ops32bit
text    data     bss     dec     hex filename
62632     432    7084   70148   11204 ./obj/main/cleanflight_CJMCU.elf

ReverseBits5logNOps
text    data     bss     dec     hex filename
62608     436    7084   70128   111f0 ./obj/main/cleanflight_CJMCU.elf

current bitReverse
text    data     bss     dec     hex filename
62592     432    7084   70108   111dc ./obj/main/cleanflight_CJMCU.elf

so current implementation has smallest ROM size. According to link it is slightly slower, but I would say ROM size is the priority here, since NRF24L01 is most likely to be used on small brushed quads.